### PR TITLE
cmd/atlas: diff printing

### PR DESCRIPTION
**Pull request by rotemtam**
image

### CODE REVIEW

**Review by yonidavidson — APPROVED** [review]
LGTM,
updated the text to be a valid migration with explanations.

File: cmd/atlas/apply.go [hunk 1]
```diff
@@ -58,6 +58,15 @@ func applyRun(d *Driver, u schemaUnmarshaler, dsn string, file string) {
 	cobra.CheckErr(err)
 	changes, err := d.SchemaDiff(s, &desired)
 	cobra.CheckErr(err)
+	schemaCmd.Println("-- Planned Changes:")
+	for _, ch := range changes {
+		desc, err := changeDescriptor(ctx, ch, d)
+		cobra.CheckErr(err)
+		schemaCmd.Println("--",desc.typ, ":", desc.subject)
```
Comment (a8m) [inline cmd/atlas/apply.go:65]: Is it go formatted?

File: cmd/atlas/apply.go [hunk 2]
```diff
@@ -58,6 +58,15 @@ func applyRun(d *Driver, u schemaUnmarshaler, dsn string, file string) {
...
+		for _, q := range desc.queries {
+			schemaCmd.Println( q,";")
```
Comment (a8m) [inline cmd/atlas/apply.go:67]: ditto

File: cmd/atlas/mux.go [hunk 3]
```diff
@@ -112,3 +115,32 @@ func (p *schemaMarshal) marshal(s *schema.Schema) ([]byte, error) {
...
+func (i *interceptor) on() {
+	i.intercept = true
+}
+
+func (i *interceptor) off() {
+	i.intercept = false
+}
+
+func (i *interceptor) clear() {
+	i.history = []string{}
```
Comment (a8m) [inline cmd/atlas/mux.go:137]:
```suggestion
	i.history = nil
```

**Review by a8m — CHANGES_REQUESTED** [review]

**Review by a8m — APPROVED** [review]

**Pull request merged**